static/merge.js

```diff
@@ -0,0 +1,325 @@
+(() => {
+    // Horizontal sublevels which were merged.
+    var mergedHorizontalLevels = {};
+
+    // Describes in what position new elements are added to table.
+    var startIndex = 0;
+
+    /**
+     * Adds new column to table.
+     * @param {Array} star_amount - Amount of observation per student.
+     * @param {Integer} startIndex - Position inside table where new elements are created.
+     * @param {Integer} constructId - ID of construct
+     * @param {Integer} i - Index element.
+     * @param {Boolean} isLast - Describes if level was last in table.
+     */
+    function addColumn(star_amount, startIndex, constructId, i, isLast) {
+        if (star_amount.length !== 0) {
+            addColumnWithStar(startIndex, constructId, star_amount, i, isLast)
+        } else {
+            isLast 
+                ? addEmptyCellAfter(constructId, i + 1, startIndex - 2)
+                : addEmptyCellBefore(constructId, i + 1, startIndex - 1);
+        }
+    }
+
+    function addColumnWithStar(startIndex, constructId, stars, rowIndex, isLast) {
+        let span = createSpan(stars);
+
+        isLast
+            ? starColumnAfter(rowIndex + 1, constructId, startIndex - 2, stars, span)
+            : starColumnBefore(rowIndex + 1, constructId, startIndex - 1, stars, span);
+    }
+
+    /**
+     * Check if level is last inside table.
+     * Level is last when sum of his `startIndex` and `<th>` elements is
+     * equal to all headers inside table - cells for students name(Those are maked with <th> too)
+     * and - count of levels for construct.
+     * @param {Integer} constructId 
+     * @param {String} construct 
+     * @param {Array} headers 
+     */
+    function isLastLevel(constructId, construct, headers, levelCount) {
+        const cellIndex = headers[0].cellIndex;
+
+        const cells = $(`.star_chart_table-${constructId} .${construct}`)
+            .find(`td:eq(0)`);
+        startIndex = cellIndex;
+        const allHeaders = $(`.star_chart_table-${constructId}`)
+            .find('th');
+    
+        return startIndex + (headers.length) === allHeaders.length - cells.length - levelCount - 1 ?
```

Comment:
This statement is definitely too long and you do here too many things at once. BTW:
```
isLast = startIndex + (headers.length) === allHeaders.length - cells.length - levelCount - 1
```